components/merkledb/benches/storage.rs

```diff
@@ -232,6 +232,27 @@ fn proof_list_append(b: &mut Bencher, len: usize) {
     );
 }
 
+fn proof_list_extend(b: &mut Bencher, len: usize) {
```

Comment:
May you attach benchmark results?

Reply:
Where to? Here are some results from my laptop:

### New implementation

```text
storage/proof_list/extend/items/1000
        time:   [3.2060 ms 3.2233 ms 3.2460 ms]
        thrpt:  [308.07 Kelem/s 310.24 Kelem/s 311.92 Kelem/s]
storage/proof_list/extend/items/10000
        time:   [35.507 ms 35.960 ms 36.368 ms]
        thrpt:  [274.97 Kelem/s 278.09 Kelem/s 281.64 Kelem/s]
storage/proof_list/extend/items/100000
        time:   [379.63 ms 381.62 ms 384.91 ms]
        thrpt:  [259.80 Kelem/s 262.04 Kelem/s 263.42 Kelem/s]
```

### Old implementation

```text
storage/proof_list/extend/items/1000
        time:   [14.093 ms 14.502 ms 14.857 ms]
        thrpt:  [67.308 Kelem/s 68.955 Kelem/s 70.956 Kelem/s]
  change:
        time:   [+336.65% +354.39% +378.13%] (p = 0.00 < 0.05)
        thrpt:  [-79.085% -77.992% -77.099%]
storage/proof_list/extend/items/10000
        time:   [204.36 ms 206.86 ms 209.83 ms]
        thrpt:  [47.657 Kelem/s 48.341 Kelem/s 48.934 Kelem/s]
  change:
        time:   [+466.72% +481.71% +500.94%] (p = 0.00 < 0.05)
        thrpt:  [-83.359% -82.809% -82.354%]
storage/proof_list/extend/items/100000
        time:   [3.1680 s 3.3873 s 3.5454 s]
        thrpt:  [28.206 Kelem/s 29.522 Kelem/s 31.565 Kelem/s]
  change:
        time:   [+656.77% +716.51% +772.67%] (p = 0.00 < 0.05)
        thrpt:  [-88.541% -87.753% -86.786%]
```